Exception related to cuda when enumerating openvino compatible backends

Starting with commit https://github.com/opencv/opencv/commit/194905642384ade117591237ca2c497419550390
i've been getting the following exception when calling `cv::dnn::getAvailableTargets(cv::dnn::DNN_BACKEND_INFERENCE_ENGINE)`

> exception: OpenCV(4.5.0-pre) D:\Dev\opencv\modules\dnn\src\cuda4dnn/init.hpp:54: error: (-217:Gpu API call) CUDA driver version is insufficient for CUDA runtime version in function 'cv::dnn::cuda4dnn::getDevice'

I reverted the aforementioned commit and it works again
This happens with OpenCV built with cudnn support but run on a computer without Nvidia graphic card.
The reason why we enter `int cv::dnn::cuda4dnn::getDevice()` when enumerating targets for a non-cuda backend is because of `isDeviceCompatible` call here:
https://github.com/opencv/opencv/blob/45ee8e25329aedcc8c23da819abf4f1800e67e56/modules/dnn/src/dnn.cpp#L232

--- a/modules/dnn/src/cuda4dnn/init.hpp
+++ b/modules/dnn/src/cuda4dnn/init.hpp
@@ -57,7 +57,7 @@ namespace cv { namespace dnn { namespace cuda4dnn {
 
     bool isDeviceCompatible()
     {
-        if (cv::cuda::getCudaEnabledDeviceCount() <= 0)
+        if (getDeviceCount() <= 0)
             return false;
 
         int device_id = getDevice();
@@ -80,7 +80,7 @@ namespace cv { namespace dnn { namespace cuda4dnn {
 
     bool doesDeviceSupportFP16()
     {
-        if (cv::cuda::getCudaEnabledDeviceCount() <= 0)
+        if (getDeviceCount() <= 0)
             return false;
 
         int device_id = getDevice();

--- a/modules/dnn/src/cuda4dnn/init.hpp
+++ b/modules/dnn/src/cuda4dnn/init.hpp
@@ -57,6 +57,9 @@ namespace cv { namespace dnn { namespace cuda4dnn {
 
     bool isDeviceCompatible()
     {
+        if (cv::cuda::getCudaEnabledDeviceCount() <= 0)
+            return false;
+
         int device_id = getDevice();
         if (device_id < 0)
             return false;
@@ -77,6 +80,9 @@ namespace cv { namespace dnn { namespace cuda4dnn {
 
     bool doesDeviceSupportFP16()
     {
+        if (cv::cuda::getCudaEnabledDeviceCount() <= 0)
+            return false;
+
         int device_id = getDevice();
         if (device_id < 0)
             return false;

